staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -580,6 +580,40 @@ type PatchOptions struct {
 	FieldManager string `json:"fieldManager,omitempty" protobuf:"bytes,3,name=fieldManager"`
 }
 
+// ApplyOptions may be provided when applying an API object.
+// ApplyOptions is meant to be a subset of PatchOptions that does not include FieldManager.
+// FieldManager is required for apply requests so it is required parameter on clientset
+// Apply functions, and therefore not included in ApplyOptions.
+type ApplyOptions struct {
+	TypeMeta `json:",inline"`
+
+	// When present, indicates that modifications should not be
+	// persisted. An invalid or unrecognized dryRun directive will
+	// result in an error response and no further processing of the
+	// request. Valid values are:
+	// - All: all dry run stages will be processed
+	// +optional
+	DryRun []string `json:"dryRun,omitempty" protobuf:"bytes,1,rep,name=dryRun"`
+
+	// Force is going to "force" Apply requests. It means user will
+	// re-acquire conflicting fields owned by other people. Force
+	// flag must be unset for non-apply patch requests.
+	Force bool `json:"force" protobuf:"varint,2,opt,name=force"`
+
+	// fieldManager is a name associated with the actor or entity
+	// that is making these changes. The value must be less than or
+	// 128 characters long, and only contain printable characters,
+	// as defined by https://golang.org/pkg/unicode/#IsPrint. This
+	// field is required for apply requests
+	// (application/apply-patch) but optional for non-apply patch
```

Comment:
Since we're in applyoptions, this should be mandatory for all requests I think?

Reply:
Yes, good point. I'll just trim this up to only say what is relevant about apply, I'll do the same for force.